PropertyChanged.Fody.Analyzer/SourceGenerator.cs

```diff
@@ -0,0 +1,102 @@
+﻿using System.Collections.Immutable;
+using System.Diagnostics;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Diagnostics;
+
+[Generator(LanguageNames.CSharp)]
+public class SourceGenerator : IIncrementalGenerator
+{
+    public void Initialize(IncrementalGeneratorInitializationContext context)
+    {
+        var classesProvider = context.SyntaxProvider.CreateSyntaxProvider(IsCandidateForGenerator, GetSyntaxForCandidate).ExceptNullItems();
+
+        var input = context.CompilationProvider
+            .Combine(context.AnalyzerConfigOptionsProvider)
+            .Combine(classesProvider.Collect())
+            .Select((((Compilation, AnalyzerConfigOptionsProvider), ImmutableArray<ClassDeclarationSyntax>) args, CancellationToken _) =>
+            {
+                var ((compilation, options), classes) = args;
+
+                options.GlobalOptions.TryGetValue("build_property.PropertyChanged_GeneratorConfiguration", out var configurationSource);
+
+                var configuration = Configuration.Read(configurationSource);
+
+                return (compilation, configuration, classes);
+            });
+
+        context.RegisterSourceOutput(input, GenerateSource);
+    }
+
+    static void GenerateSource(SourceProductionContext context, (Compilation compilation, Configuration configuration, ImmutableArray<ClassDeclarationSyntax> classes) parameters)
+    {
+        var (compilation, configuration, classes) = parameters;
+
+        // Log("GenerateSource: " + string.Join(", ", classes.Select(c => c.Identifier.Text)));
+
+        SourceGeneratorEngine.GenerateSource(context, compilation, configuration, classes);
+    }
+
+    static bool IsCandidateForGenerator(SyntaxNode syntaxNode, CancellationToken token)
+    {
+        try
+        {
+            return syntaxNode is ClassDeclarationSyntax { Parent: not ClassDeclarationSyntax } classDeclaration
```

Comment:
Do we want to support records?

Reply:
Maybe in a later version if s.o. requests that? I don't see a real use case.

Reply:
Yup 👍

Reply:
Well, that [didn't take long](https://github.com/Fody/PropertyChanged/issues/863#issuecomment-1238475604) 😅